test/custom-service-client-options/generated/Xml/lib/autoRestSwaggerBATXMLServiceContext.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ *
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ * Changes may cause incorrect behavior and will be lost if the code is
+ * regenerated.
+ */
+
+import * as msRest from "@azure/ms-rest-js";
+import * as Models from "./models";
+
+const packageName = "";
+const packageVersion = "";
+
+export class AutoRestSwaggerBATXMLServiceContext extends msRest.ServiceClient {
+
+  /**
+   * Initializes a new instance of the AutoRestSwaggerBATXMLServiceContext class.
+   * @param [options] The parameter options
+   */
+  constructor(options?: Models.AutoRestSwaggerBATXMLServiceOptions) {
+
+    if (!options) {
+      options = {};
+    }
+
+    if (!options.userAgent) {
+      const defaultUserAgent = msRest.getDefaultUserAgentValue();
+      options.userAgent = `${packageName}/${packageVersion} ${defaultUserAgent}`;
+    }
+
+    if (true) {
+      options = {
+        ...options,
+        "noRetryPolicy": "true",
```

Comment:
Definitely agree that this should be reduced to a boolean instead of a string. We may need to have a mapping in the generator that knows the types of the ServiceClientOptions properties. If a value is provided on the command line but there isn't a known property in the mapping, then I'd say default it to a string.